ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Client.java

```diff
@@ -112,6 +112,48 @@
    */
   CompletableFuture<Void> terminatePushQuery(String queryId);
 
+  /**
+   * Sends a SQL request to the ksqlDB server. This method supports 'CREATE', 'CREATE ... AS
+   * SELECT', 'DROP', 'TERMINATE', and 'INSERT INTO ... AS SELECT' statements.
+   *
+   * <p>Each request should contain exactly one statement. Requests that contain multiple statements
+   * will be rejected by the client, in the form of failing the {@code CompletableFuture}, and the
+   * request will not be sent to the server.
+   *
+   * <p>The {@code CompletableFuture} is completed once a response is received from the server.
+   * Note that the actual execution of the submitted statement is asynchronous, so the statement
```

Comment:
As explained in the javadocs, the CompletableFuture is completed once the server accepts the statement. The actual statement execution is async, so the statement likely will not have been executed by the time the future is complete. Hopefully this is not too confusing to users.

Reply:
I'm curious why we don't complete the CompletableFuture when the statement has been executed? I think that would be more intuitive for the user.
Is there any other way for the user to know when the statements has been executed? I'm thinking of use cases where they want to create a stream then start sending messages to it. Before sending the messages they need to know it's been created.

Reply:
Statement execution is async (command runner thread needs to be read the statement from the command topic, then execute), and there's no good way to know when the command has been executed. If we know the type of statement being executed, we could try to list streams/tables/queries and check the result contains the expected stream/table/query, but this feels rather hacky. It also doesn't address the problem that there may be multiple ksqlDB servers in the cluster, and the statement having been executed on one server does not guarantee that the other servers have executed the statement.

Reply:
In terms of how the user can know if the statement has been executed, besides listing streams/tables/queries as mentioned above, they can also try opening the connection to stream rows or insert new rows. That request will fail if the statement hasn't been executed yet, so they could retry the statement until it succeeds? Definitely not great but I'm not sure what better options we have.

Reply:
One correction to my earlier statement which mitigates some of the async execution pain: After reading the code more carefully, I found out that

> The actual statement execution is async, so the statement likely will not have been executed by the time the future is complete.

is not true. Under normal circumstances, the statement typically *has* executed by the time the future is complete. It only wouldn't if (1) the command runner thread was stuck for some reason, or (2) the config value for how long the server waits for the command to execute before returning a response from the /ksql endpoint was tuned to be too low.